src/streaming/sink.creator.hh

```diff
@@ -0,0 +1,183 @@
+#pragma once
+
+#include "zarr.stream.hh" // ZarrDimension_s
+#include "sink.hh"
+#include "thread.pool.hh"
+#include "s3.connection.hh"
+
+#include <optional>
+#include <memory>
+#include <unordered_map>
+
+namespace zarr {
+using Dimension = ZarrDimension_s;
+
+class SinkCreator final
```

Comment:
I previously raised this concern (and I think it was in the context of the `SinkCreator` too), but I believe we should avoid using `final`. If you disagree, let me know why. Generally, this is strongly discouraged unless there's a very good reason to use it.